polaris/dataset/_dataset.py

```diff
@@ -140,24 +147,12 @@ def _validate_model(cls, m: "Dataset"):
                 m.annotations[c] = ColumnAnnotation()
             m.annotations[c].dtype = m.table[c].dtype
 
-        # Verify the checksum
-        # NOTE (cwognum): Is it still reasonable to always verify this as the dataset size grows?
-        actual = m.md5sum
-        expected = cls._compute_checksum(m.table)
-
-        if actual is None:
-            m.md5sum = expected
-        elif actual != expected:
-            raise PolarisChecksumError(
-                "The dataset md5sum does not match what was specified in the meta-data. "
-                f"{actual} != {expected}"
-            )
-
         # Set the default cache dir if none and make sure it exists
         if m.cache_dir is None:
-            m.cache_dir = fs.join(DEFAULT_CACHE_DIR, _CACHE_SUBDIR, m.name, m.md5sum)
-        fs.mkdir(m.cache_dir, exist_ok=True)
+            dataset_id = m._md5sum if m.has_md5sum else str(uuid.uuid4())
```

Comment:
Is there a public getter for `_md5sum`?

Reply:
Given that we're accessing the private property from within the same class, I am okay with this access!